tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Generators/ReadMeGeneratorTool.cs

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+using Azure.AI.OpenAI;
+using Azure.Sdk.Tools.Cli.Commands;
+using Azure.Sdk.Tools.Cli.Contract;
+using Azure.Sdk.Tools.Cli.Services;
+using OpenAI.Chat;
+
+namespace Azure.Sdk.Tools.Cli.Tools
+{
+    // Will add after we decide which tools are exported via the MCP.
+    //[McpServerToolType, Description("Generates a README file, using service documentation")]
+    public partial class ReadMeGeneratorTool : MCPTool
+    {
+        private readonly ILogger<ReadMeGeneratorTool> logger;
+        private readonly IOutputService output;
+        private readonly AzureOpenAIClient openAiClient;
+        private Option<string> packagePathOption;
+        private Option<string> outputPathOption;
+        private Option<string> templatePathOption;
+        private Option<string> serviceDocumentationOption;
+        private Option<string> modelOption;
+
+        public ReadMeGeneratorTool(ILogger<ReadMeGeneratorTool> logger, IOutputService output, AzureOpenAIClient openAiClient)
+        {
+            ArgumentNullException.ThrowIfNull(logger, nameof(logger));
+            ArgumentNullException.ThrowIfNull(output, nameof(output));
+
+            // This should be removed, as it won't be possible to have `null` here when the
+            // patterns for DI are fixed.
+            ArgumentNullException.ThrowIfNull(openAiClient, nameof(openAiClient));
+            this.logger = logger;
+            this.output = output;
+            this.openAiClient = openAiClient;
+
+            this.CommandHierarchy = [SharedCommandGroups.Generators];
+        }
+
+        public override Command GetCommand()
+        {
+            templatePathOption = new Option<string>("--template-path")
+            {
+                IsRequired = true,
+                Description = "Path to the README template file (ie: Templates/README-template.go.md)",
+            };
+
+            templatePathOption.SetDefaultValue("Templates/ReadMeGenerator/README-template.go.md");
+
+            serviceDocumentationOption = new Option<string>(
+                "--service-url")
+            {
+                Description = "URL to the service documentation (ex: https://learn.microsoft.com/en-us/azure/service-bus-messaging)",
```

Comment:
```suggestion
                Description = "URL to the service documentation (ex: https://learn.microsoft.com/azure/service-bus-messaging)",
```

Lest the link checker flag us.